### PR TITLE
[WIP] Add LSP support for Lua (EmmyLua-LS-all).

### PR DESCRIPTION
**Pull request by sunlin7**
Add the LSP support for Lua (EmmyLua-LS-all), and it's easy to extend with other Lua LSP backend.

### CODE REVIEW

**Comment by sunlin7** [comment]
Hi @syl20bnr ,
Could you please review and merge this PR? Thank you.

File: layers/+lang/lua/packages.el [hunk 1]
```diff
@@ -30,17 +30,10 @@
...
     :init
     (progn
-      (setq lua-indent-level 2
-            lua-indent-string-contents t)
-      (spacemacs/set-leader-keys-for-major-mode 'lua-mode
-        "d" 'lua-search-documentation
-        "sb" 'lua-send-buffer
-        "sf" 'lua-send-defun
-        "sl" 'lua-send-current-line
-        "sr" 'lua-send-region))))
```
Comment (smile13241324) [inline layers/+lang/lua/packages.el]: You should add a binding to "'" to start a repl process as well as add a repl registration like in 
```elisp
      (spacemacs/register-repl 'python
                               'spacemacs/python-start-or-switch-repl "python")
```
For reference you should check the python layer.

**Comment by smile13241324** [comment]
CI is strange, can you rebase develop on your branch and push again? That's what CI complains about right now.

**Comment by sunlin7** [comment]
@smile13241324 thanks for your comment, I'll change them then update again soon.

**Comment by sunlin7** [comment]
Hi @smile13241324 , I had modify the code according your comments in my local.
Emm..., I have to re-create an new pull request for my fork was deleted by me last week. So I  fork the repo and recall modifications then create pull request #12991 with last modifications. 
Please review #12991, and I close this pull request. Apologize for my mistake.

**Pull request closed without merging**